test/LEGO.AsyncAPI.Tests/Models/AvroSchema_Should.cs

```diff
@@ -0,0 +1,341 @@
+                        Doc = "A list of email addresses.",
+                    },
+                    new AvroField
+                    {
+                        Name = "metadata",
+                        Type = new AvroMap
+                        {
+                            Values = AvroPrimitiveType.String,
+                        },
+                        Doc = "Metadata associated with the user.",
+                    },
+                    new AvroField
+                    {
+                        Name = "address",
+                        Type = new AvroRecord
+                        {
+                            Name = "Address",
+                            Fields = new List<AvroField>
+                            {
+                                new AvroField { Name = "street", Type = AvroPrimitiveType.String },
+                                new AvroField { Name = "city", Type = AvroPrimitiveType.String },
+                                new AvroField { Name = "zipcode", Type = AvroPrimitiveType.String },
+                            },
+                        },
+                        Doc = "The address of the user.",
+                    },
+                    new AvroField
+                    {
+                        Name = "profilePicture",
+                        Type = new AvroFixed
+                        {
+                            Name = "ProfilePicture",
+                            Size = 256,
+                        },
+                        Doc = "A fixed-size profile picture.",
+                    },
+                    new AvroField
+                    {
+                        Name = "contact",
+                        Type = new AvroUnion
+                        {
+                            Types = new List<AvroFieldType>
+                            {
+                                AvroPrimitiveType.Null,
+                                new AvroRecord
+                                {
+                                    Name = "PhoneNumber",
+                                    Fields = new List<AvroField>
+                                    {
+                                        new AvroField { Name = "countryCode", Type = AvroPrimitiveType.Int },
+                                        new AvroField { Name = "number", Type = AvroPrimitiveType.String },
+                                    },
+                                },
+                            },
+                        },
+                        Doc = "The contact information of the user, which can be either null or a phone number.",
+                    },
+                },
+            };
+
+            // Act
+            var actual = schema.SerializeAsYaml(AsyncApiVersion.AsyncApi2_0);
+
+            // Assert
+            actual.Should()
+                  .BePlatformAgnosticEquivalentTo(expected);
+        }
+
+        [Test]
+        public void ReadFragment_DeserializesCorrectly()
```

Comment:
Do we need any testing around `.avsc` configured in JSON format? For example events that are $ref'd like so:
```json
{
  "type": "record",
  "name": "SomeEvent",
  "namespace": "my.namspace.for.event",
  "example": {
    "occurredOn": "2023-11-03T09:59:56.582908743Z",
    "partnerId": "1",
    "platformSource": "Brecht",
    "countryCode": "DE"
  },
  "fields": [
    {
      "name": "countryCode",
      "type": "string",
      "doc": "Country of the partner, (e.g. DE)"
    },
    {
      "name": "occurredOn",
      "type": "string",
      "doc": "Timestamp of when action occurred."
    },
    {
      "name": "partnerId",
      "type": "string",
      "doc": "Id of the partner"
    },
    {
      "name": "platformSource",
      "type": "string",
      "doc": "Platform source"
    }
  ]
}
```

Reply:
full support for references and custom fields as part of the metadata dictionary.